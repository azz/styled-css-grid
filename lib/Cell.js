import styled from "styled-components";
import PropTypes from "prop-types";

const Cell = styled.section`
  height: 100%;
  min-width: 0;
  align-content: space-around;
  grid-column-end: ${({ width = 1 }) => `span ${width}`};
  grid-row-end: ${({ height = 1 }) => `span ${height}`};

  ${({ left }) => left && `grid-column-start: ${left}`};
  ${({ top }) => top && `grid-row-start: ${top}`};

  ${({ center }) => center && `text-align: center`};

  ${({ area }) => area && `grid-area: ${area}`};

  ${/* prettier-ignore */
  ({ middle }) => middle && `
    display: inline-flex;
    flex-flow: column wrap;
    justify-content: center;
    justify-self: stretch;
  `};
`;

Cell.propTypes = {
  className: PropTypes.string,
  width: PropTypes.number,
  height: PropTypes.number,
  top: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  left: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  middle: PropTypes.bool,
  center: PropTypes.bool,
  area: PropTypes.string
};

export default Cell;
