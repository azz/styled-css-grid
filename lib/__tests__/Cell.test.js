import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import Cell from "../Cell";

describe("<Cell/>", () => {
  test("matches snapshot with default args", () => {
    const tree = renderer.create(<Cell />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("'width' prop sets 'grid-column-end'", () => {
    const tree = renderer.create(<Cell width={3} />).toJSON();
    expect(tree).toHaveStyleRule("grid-column-end", "span 3");
  });

  test("'height' prop sets 'grid-row-end'", () => {
    const tree = renderer.create(<Cell height={3} />).toJSON();
    expect(tree).toHaveStyleRule("grid-row-end", "span 3");
  });

  test("'middle' prop vertically aligns contents", () => {
    const tree = renderer.create(<Cell middle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("'center' prop horizontally aligns contents", () => {
    const tree = renderer.create(<Cell center />).toJSON();
    expect(tree).toHaveStyleRule("text-align", "center");
  });

  test("'area' prop sets 'grid-area'", () => {
    const tree = renderer.create(<Cell area="a" />).toJSON();
    expect(tree).toHaveStyleRule("grid-area", "a");
  });
});
