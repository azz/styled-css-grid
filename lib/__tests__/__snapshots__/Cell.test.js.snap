// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell/> 'middle' prop vertically aligns contents 1`] = `
.c0 {
  height: 100%;
  min-width: 0;
  overflow: hidden;
  -webkit-align-content: space-around;
  -ms-flex-line-pack: space-around;
  align-content: space-around;
  grid-column-end: span 1;
  grid-row-end: span 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-box-packjustify-self: stretch;
  -webkit-justify-self: stretch;
  -ms-flex-packjustify-self: stretch;
  justify-self: stretch;
}

<section
  className="c0"
/>
`;

exports[`<Cell/> matches snapshot with default args 1`] = `
.c0 {
  height: 100%;
  min-width: 0;
  overflow: hidden;
  -webkit-align-content: space-around;
  -ms-flex-line-pack: space-around;
  align-content: space-around;
  grid-column-end: span 1;
  grid-row-end: span 1;
}

<section
  className="c0"
/>
`;
